### PR TITLE
[common/catalog] refactor: move TableSnapshot into crate common/catalog, so that a plan could have a reference to table info

### PR DESCRIPTION
**Pull request by drmingdrmer**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Changelog

- Improvement

## Related Issues

- #2055 
- #2059
- fix: #2060

### CODE REVIEW

File: common/catalog/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+publish = false
+edition = "2021"
+
+[dependencies] # In alphabetical order
+# Workspace dependencies
+common-base = {path = "../base"}
+common-arrow = {path = "../arrow"}
+common-datavalues = {path = "../datavalues"}
+#common-exception = {path = "../exception"}
+#common-io = {path = "../io"}
```
Comment (leiysky) [inline common/catalog/Cargo.toml:15]: Shall we remove these lines?

Comment (drmingdrmer) [reply common/catalog/Cargo.toml:15]: Oh my...sure

**Pull request merged**